libsolutil/CommonIO.cpp

```diff
@@ -132,18 +132,11 @@ int solidity::util::readStandardInputChar()
 
 string solidity::util::absolutePath(string const& _path, string const& _reference)
 {
-	boost::filesystem::path p(_path);
-	// Anything that does not start with `.` is an absolute path.
-	if (p.begin() == p.end() || (*p.begin() != "." && *p.begin() != ".."))
-		return _path;
-	boost::filesystem::path result(_reference);
-	result.remove_filename();
-	for (boost::filesystem::path::iterator it = p.begin(); it != p.end(); ++it)
-		if (*it == "..")
-			result = result.parent_path();
-		else if (*it != ".")
-			result /= *it;
-	return result.generic_string();
+	// TODO: Handle throw of boost::filesystem::filesystem_error
```

Comment:
Yeah, looks like `canonical()` differs from our implementation in that it requires paths to actually exist in the filesystem. That's probably a good thing but I guess it might require adjustments in `CompilerStack::loadMissingSources()` due to path remapping.